highlighters/main/test-data/back-quoted-argument.zsh

```diff
@@ -27,16 +27,25 @@
 # vim: ft=zsh sw=2 ts=2 et
 # -------------------------------------------------------------------------------------------------
 
-# 42 is in the command position in a nested subshell.
 BUFFER='echo `echo \`42\`` "is `echo equal` to" `echo 6 times 9'
 
 expected_region_highlight=(
   "1 4 builtin" # echo
   "6 18 default" # `echo \`42\``
   "6 18 back-quoted-argument" # `echo \`42\``
+  "7 10 builtin" # echo
+  "12 17 default" # \`42\`
+  "12 17 back-quoted-argument" # \`42\`
+  "14 15 unknown-token" # 42
   "20 39 default" # "is `echo equal` to"
   "20 39 double-quoted-argument" # "is `echo equal` to"
   "24 35 back-quoted-argument" # `echo equal`
+  "25 28 builtin" # echo
+  "30 34 default" # equal
   "41 55 default" # `echo 6 times 9
```

Comment:
Should the backticks themselves be highlighted?  Quotes and command/process substitutions are all highlighted.  I suppose we could highlight backticks as magenta, like `$(…)`, since both of these are subject to word splitting.

Reply:
That wouldn't be consistent with single, double, and dollar quotes, but then again those are quotes and these are command substitutions.

Reply:
_What_ wouldn't be consistent? The colour (yellow v. magenta), or something else?

Reply:
For the quoting styles, the quotes themselves are not given a special style. If `$(` and `)` and backticks are, that would be inconsistent between the two; but they are different syntactical constructs. Additionally I think magenta (or any other really) highlighting all the words not otherwise highlighted in a command substitution is a bit much.

Reply:
Agreed that highlighting all words inside a cmdsubst as magenta is a bit much.  How about highlighting `$(`, `)`, `` ` `` as `command-substitution-delimiter`, defaulting to magenta, and keeping plain words inside the cmdsubst unhighlighted?  We could even keep the `command-substitution` group matching the entire `$(…)` construct so people could assign to it in themes.

An orthogonal question: if we had `command-substitution-{delimiter-,}$(( N++ ))` styles, with $N being the nesting depth, we could have a [rainbow parentheses](https://github.com/luochen1990/rainbow) theme.  Featuritis?

Reply:
I was thinking we'd consider that when coming up with zstyles for #362 . I like the delimiter idea. How about the latest two commits?

Note: `: "$(: foo)"` will still have foo highlighted yellow since there's the double-quoted-argument highlighting applying to the whole of the double quotes. Not sure what if anything to do about that.

Reply:
Delimiters LGTM.

Re `foo` in `: "$(: foo)"`, I don't think it should be yellow since the lexing rules of double-quoted strings (e.g., backslash escape sequences) don't apply there. That's not to say whether or not it blocks merging or releasing; just to say that I don't think it's useful for that `foo` to be yellow.

Reply:
I agree, but I don't know what to do about it. In theory
`: ${ZSH_HIGHLIGHT_STYLES[default]:=fg=default}`
would help, but it appears to make no difference here. Additionally the yellow would still show through for `path` styled strings.